src/constructors/regular.c

```diff
@@ -931,3 +931,36 @@ igraph_error_t igraph_extended_chordal_ring(
     IGRAPH_FINALLY_CLEAN(1);
     return IGRAPH_SUCCESS;
 }
+
+/**
+ * \function igraph_hypercube
+ * \brief Creates an n-dimensional hypercube.
+ * 
+ * \param graph Pointer to an uninitialized graph object.
+ * \param nodes Integer constant, n-dimensional hypercube, 2^n vertices.
+ * \param directed Whether the graph should be directed.
+ * \return Error code.
+ */
+igraph_error_t igraph_hypercube(
+    igraph_t *graph, igraph_integer_t nodes, igraph_bool_t directed) {
+
+    igraph_vector_int_t dims;
+    igraph_integer_t i;
+
+    /* Calculate dimension vector for the hypercube */    
+    igraph_vector_int_init(&dims, nodes);
+    igraph_vector_int_fill(&dims, 2);
+
+    /* Use igraph_square_lattice to construct hypercube */
+    igraph_error_t status = 
+    igraph_square_lattice(graph, &dims, 1, directed, 0, 0);
```

Comment:
Instead of returning the exit code of other igraph functions, simply wrap them in `IGRAPH_CHECK`.

```suggestion
    IGRAPH_CHECK(igraph_square_lattice(graph, &dims, 1, directed, false, NULL));
```

Also, please prefer `NULL` (instead of 0) for null pointers and `true` / `false` (instead of 1/0) for boolean values. This is more readable.